### PR TITLE
Add condition to the Front Page Section container

### LINKED ISSUE #1811: Do not use a list if there is just one StoryPromo
**Is your feature request related to a problem? Please describe.**
At the moment, in the `FrontPageSection` container we are displaying the `StoryPromo` as a list item even if there is just one in the group items array.

**Describe the solution you'd like**
Do not wrap the `StoryPromo` in a list if the items array has just one item.

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: snapshot and unit tests

**Additional context**

### PR DESCRIPTION
**Pull request by DenisHdz**
Resolves #1811

**Overall change:** 
Add a condition to check if the group items array has one or more items to display just a promo or a list of promos.

**Code changes:**
- Add a ternary operator in the `FrontPageSection` container to display either the `StoryPromoList` or just a `StoryPromo`.

---

- [X] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/containers/FrontPageSection/index.jsx [hunk 1]
```diff
@@ -36,13 +36,17 @@ const FrontPageSection = ({ bar, group }) => {
...
+      {items.length > 1 ? (
+        <StoryPromoUl>
+          {items.map(item => (
+            <StoryPromoLi key={item.id}>
+              <StoryPromo item={item} script={script} />
+            </StoryPromoLi>
+          ))}
+        </StoryPromoUl>
+      ) : (
+        items.length === 1 && <StoryPromo item={items[0]} script={script} />
```
Comment (andrew-nowak) [inline src/app/containers/FrontPageSection/index.jsx]: Do you need this condition? This case should be handled by the `items.length === 0` check above, shouldn't it?

Comment (ghost) [inline src/app/containers/FrontPageSection/index.jsx]: Do we need another check here?

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by andrew-nowak — CHANGES_REQUESTED** [review]
should have a test case for not rendering the `<ul>` and `<li>` when only one item

File: src/app/containers/FrontPageSection/index.jsx (hunk 1 above)
Comment (DenisHdz) [reply src/app/containers/FrontPageSection/index.jsx]: Nope, we don't. It is already checked at the top.

**Review by andrew-nowak — DISMISSED** [review]
👍

File: src/app/containers/FrontPageSection/index.test.jsx [hunk 2]
```diff
@@ -195,5 +223,16 @@ describe('FrontPageSection Container', () => {
       // IF THERE WERE ANY!
       expect(container.children).toHaveLength(0);
     });
+
+    it('should not render the ul and li when only one item', () => {
```
Comment (chris-hinds) [inline src/app/containers/FrontPageSection/index.test.jsx]:
```suggestion
    it('should not render the story promo inside a list when only one item exists', () => {
```

**Review by chris-hinds — COMMENTED** [review]
Just one tiny comment on the wording of the test case. 👍

**Review by andrew-nowak — DISMISSED** [review]

**Review by chris-hinds — DISMISSED** [review]

**Review by ghost — CHANGES_REQUESTED** [review]
I think we also need to have a snapshot test showing the expected structure when only one item is present. Currently we only have snaps for groups with more than one item.

**Review by ghost — APPROVED** [review]
Great Work 👍

**Review by andrew-nowak — APPROVED** [review]
still looks great :+1:

**Comment by david-boydell** [comment]
Hi Denis, I cannot seem to see this on storybook. To get this to view on storybook or the routed frontpage should I add a section to the fixture data in `data/prod/pidgin/frontpage/index.json` with one entry then rebuild? If so I'll test using that.

Also if I do the above should I include the change to this ticket so that it's there for others or just leave it - not sure what we're doing with the fixture data.

Also how would I find out whether I should change `data/prod/pidgin/frontpage/index.json` or `data/test/pidgin/frontpage/index.json`? Not sure what the app/storybook is pointing to.

**Comment by DenisHdz** [comment]
> Hi Denis, I cannot seem to see this on storybook. To get this to view on storybook should I add a section to the fixture data in `data/prod/pidgin/frontpage/index.json` with one entry then rebuild? If so I'll test using that.
> 
> Also if I do the above should I include the change to this ticket so that it's there for others or just leave it - not sure what we're doing with the fixture data.
> 
> Also how would I find out whether I should change `data/prod/pidgin/frontpage/index.json` or `data/test/pidgin/frontpage/index.json`? Not sure what the app/storybook is pointing to.

At the moment we are using the `test` data in the stories but it is the same as the `prod`.
All the groups in the fixture data have more than one item, so you can not visually see this case in the Front Page. We have added a unit test for that. 
If you still want to check it, you can add a group with one item to the fixture data but I wouldn't commit it.

**Comment by david-boydell** [comment]
👍

**Comment by david-boydell** [comment]
Looks good, thanks, this is RFM.

**Pull request merged**